src/PKSim.Presentation/Presenters/Charts/EditTimeProfileAnalysisChartPresenter.cs

```diff
@@ -79,6 +89,55 @@ protected override void RefreshData()
       {
          return _chartDataCreator.CreateFor(PopulationAnalysisChart, AggregationFunctions.QuantityAggregation);
       }
+      
+      protected IEnumerable<PopulationPKAnalysis> aggregatePKAnalysis(IPopulationDataCollector populationDataCollector, IEnumerable<QuantityPKParameter> pkParameters, string captionPrefix)
+      {
+         var names = pkParameters.Select(x => x.Name).Distinct().ToList();
+         var matrix = new FloatMatrix();
+         pkParameters.Each(pkParameter => matrix.AddValuesAndSort(pkParameter.ValuesAsArray));
+
+         var results = new List<PopulationPKAnalysis>();
+         var selectedStatistics = PopulationAnalysisChart.PopulationAnalysis.SelectedStatistics;
+         var simulation = populationDataCollector as Simulation;
+         selectedStatistics.Each(statisticalAnalysis => {
+            var aggregated = _statisticalDataCalculator.StatisticalDataFor(matrix, statisticalAnalysis).ToList();
+            for (var aggregationIndex = 0; aggregationIndex < aggregated.Count; aggregationIndex++)
+            {
+               var pk = pkParameters.ElementAt(aggregationIndex);
+               var curveData = buildCurveData(pk, aggregated, aggregationIndex, statisticalAnalysis, captionPrefix);
+               results.Add(buildPopulationPKAnalysis(curveData, populationDataCollector.Compounds.First(x => simulation.Model.MoleculeNameFor(curveData.QuantityPath) == x.Name), aggregated[aggregationIndex], names, simulation));
+            }
+         });
+         return results;
+      }
+
+      private CurveData<TimeProfileXValue, TimeProfileYValue> buildCurveData(QuantityPKParameter pk, List<float[]> values, int index, StatisticalAggregation statisticalAggregation, string captionPrefix)
```

Comment:
it feels that the values and index are only use to create the caption. Maybe a method that returns the caption can be created and the caption passe dto this function insteaD? Also this is not presentation related and could be moved

Reply:
> it feels that the values and index are only use to create the caption. Maybe a method that returns the caption can be created and the caption passe dto this function insteaD? Also this is not presentation related and could be moved

It is also used to match the results from the statistical aggregation to the original source. But I did refactor this a bit for better readability.